rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -2449,21 +2458,24 @@ namespace rsx
 	//Pause/cont wrappers for FIFO ctrl. Never call this from rsx thread itself!
 	void thread::pause()
 	{
-		external_interrupt_lock.store(true);
-		while (!external_interrupt_ack.load())
+		while (UNLIKELY(external_interrupt_lock.exchange(true)))
 		{
-			if (Emu.IsStopped())
-				break;
+			std::this_thread::yield(); // TODO
+		}
 
-			_mm_pause();
+		if (!external_interrupt_ack)
+		{
+			external_interrupt_ack.wait(false);
```

Comment:
Same here. Our atomic wait is not the most lightweight from what I could glean from the impl, otherwise I'd have used it more often. Needs proper benching on windows.